app/models/user.rb

```diff
@@ -2,14 +2,18 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable, :confirmable
+         :recoverable, :rememberable, :validatable
   validates :name, presence: true
   validates :bio, presence: true
   validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   has_many :posts, foreign_key: 'author_id'
-  has_many :comments
-  has_many :likes
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def admin?
+    role == 'admin'
+  end
```

Comment:
- [ ] Kudos 💪 creating the `admin?` method inside the user model. However, in the database, we can't  😟  set the user role to `admin` because the `type` of the culomn `role` is set to `integer`. 
So you have two options to make this work: 🤗
1. If you want the role to be set to the string `admin`, you have to change the `type` of `role` to `string` from the database using a migration.
2. or simply because the role's type inside the database is set to integer and has a default of `0`, you can here assign the role to a number, for example, `1` assuming that if a user is not admin, their role is going to remain `0` and if we want them to be admin, we are going to set their role equal to `1`.  Like this: :point_down: 
```suggestion
  def admin?
    role == 1
  end
```

Reply:
hi @ridwanediallo i'll fix it right now, thank you